src/main/java/net/datafaker/transformations/SqlTransformer.java

```diff
@@ -168,19 +174,35 @@ public String generate(List<IN> input, Schema<IN, ?> schema) {
 
     @Override
     public String generate(Schema<IN, ?> schema, int limit) {
+        if (withBatchMode) {
+            return generateBatchModeStatements(schema, limit);
+        } else {
+            return generateSeparatedStatements(schema, limit);
+        }
+    }
+
+    public String generateBatchModeStatements(Schema<IN, ?> schema, int limit) {
         StringBuilder sb = new StringBuilder();
         for (int i = 0; i < limit; i++) {
             sb.append(apply(null, schema, i));
             if (i == limit - 1 && sb.length() > 0) {
-                if (withBatchMode) {
-                    sb.append(SqlDialect.getLastRowSuffix(dialect, keywordUpperCase));
-                }
+                sb.append(SqlDialect.getLastRowSuffix(dialect, keywordUpperCase));
                 sb.append(";");
             }
         }
         return sb.toString();
     }
 
+    public String generateSeparatedStatements(Schema<IN, ?> schema, int limit) {
```

Comment:
it's better to have it private otherwise people could use it and we have less freedom for refactoring

Reply:
Yep, my bad

Reply:
fixed